`native-tests.core.test-doc` fails in the second run of `pytest --ignore tests/test_bin.py`

This is a weird one. I've tested it in a fresh virtualenv.

<details>

```
$ git clean -dfx && pip install -e . && pytest --ignore tests/test_bin.py
Obtaining file:///tmp/testenv/hy
Collecting rply>=0.7.7 (from hy==0.16.0)
  Using cached https://files.pythonhosted.org/packages/e7/82/388e0845abda4bba5ba041aa32e411b3eed56f518c96c6565559e713a5ed/rply-0.7.7-py2.py3-none-any.whl
Collecting astor>=0.7.1 (from hy==0.16.0)
  Using cached https://files.pythonhosted.org/packages/35/6b/11530768cac581a12952a2aad00e1526b89d242d0b9f59534ef6e6a1752f/astor-0.7.1-py2.py3-none-any.whl
Collecting funcparserlib>=0.3.6 (from hy==0.16.0)
Collecting clint>=0.4 (from hy==0.16.0)
Collecting appdirs (from rply>=0.7.7->hy==0.16.0)
  Using cached https://files.pythonhosted.org/packages/56/eb/810e700ed1349edde4cbdc1b2a21e28cdf115f9faf263f6bbf8447c1abf3/appdirs-1.4.3-py2.py3-none-any.whl
Collecting args (from clint>=0.4->hy==0.16.0)
Installing collected packages: appdirs, rply, astor, funcparserlib, args, clint, hy
  Running setup.py develop for hy
Successfully installed appdirs-1.4.3 args-0.1.0 astor-0.7.1 clint-0.5.1 funcparserlib-0.3.6 hy rply-0.7.7
========================= test session starts ==========================
platform linux -- Python 3.7.2, pytest-4.2.1, py-1.7.0, pluggy-0.8.1
rootdir: /tmp/testenv/hy, inifile: setup.cfg
collected 579 items                                                    

tests/test_hy2py.py ..                                           [  0%]
tests/test_lex.py ...............................                [  5%]
tests/test_models.py ..............                              [  8%]
tests/compilers/test_ast.py .................................... [ 14%]
.......................................                          [ 21%] 
tests/compilers/test_compiler.py ..                              [ 21%]
tests/importer/test_importer.py ............                     [ 23%]
tests/importer/test_pyc.py .                                     [ 23%]
tests/macros/test_macro_processor.py ....                        [ 24%]
tests/macros/test_tag_macros.py .                                [ 24%]
tests/native_tests/comprehensions.hy ............                [ 26%]
tests/native_tests/core.hy ..................................... [ 32%]
...................                                              [ 36%] 
tests/native_tests/defclass.hy ...........                       [ 38%]
tests/native_tests/language.hy ................................. [ 43%]
................................................................ [ 54%] 
...............x...                                              [ 58%] 
tests/native_tests/mangling.hy ...................               [ 61%]
tests/native_tests/mathematics.hy ............................   [ 66%]
tests/native_tests/model_patterns.hy ..                          [ 66%]
tests/native_tests/native_macros.hy .......................      [ 70%]
tests/native_tests/operators.hy ................................ [ 76%]
........................                                         [ 80%] 
tests/native_tests/py35_only_tests.hy .........                  [ 81%]
tests/native_tests/py36_only_tests.hy ...                        [ 82%]
tests/native_tests/py3_only_tests.hy .......                     [ 83%]
tests/native_tests/quote.hy ..........                           [ 85%]
tests/native_tests/tag_macros.hy ..........                      [ 87%]
tests/native_tests/unless.hy .                                   [ 87%]
tests/native_tests/when.hy .                                     [ 87%]
tests/native_tests/with_decorator.hy ....                        [ 88%]
tests/native_tests/with_test.hy .....                            [ 88%]
tests/native_tests/contrib/hy_repr.hy ...........                [ 90%]
tests/native_tests/contrib/loop.hy ...                           [ 91%]
tests/native_tests/contrib/multi.hy .......                      [ 92%]
tests/native_tests/contrib/sequences.hy .....                    [ 93%]
tests/native_tests/contrib/walk.hy .........................     [ 97%]
tests/native_tests/extra/anaphoric.hy ............               [ 99%]
tests/native_tests/extra/reserved.hy .                           [100%]

================ 578 passed, 1 xfailed in 19.76 seconds ================
$ pytest --ignore tests/test_bin.py 
========================= test session starts ==========================
platform linux -- Python 3.7.2, pytest-4.2.1, py-1.7.0, pluggy-0.8.1
rootdir: /tmp/testenv/hy, inifile: setup.cfg
collected 579 items                                                    

tests/test_hy2py.py ..                                           [  0%]
tests/test_lex.py ...............................                [  5%]
tests/test_models.py ..............                              [  8%]
tests/compilers/test_ast.py .................................... [ 14%]
.......................................                          [ 21%] 
tests/compilers/test_compiler.py ..                              [ 21%]
tests/importer/test_importer.py ............                     [ 23%]
tests/importer/test_pyc.py .                                     [ 23%]
tests/macros/test_macro_processor.py ....                        [ 24%]
tests/macros/test_tag_macros.py .                                [ 24%]
tests/native_tests/comprehensions.hy ............                [ 26%]
tests/native_tests/core.hy F.................................... [ 32%]
...................                                              [ 36%] 
tests/native_tests/defclass.hy ...........                       [ 38%]
tests/native_tests/language.hy ................................. [ 43%]
................................................................ [ 54%] 
...............x...                                              [ 58%] 
tests/native_tests/mangling.hy ...................               [ 61%]
tests/native_tests/mathematics.hy ............................   [ 66%]
tests/native_tests/model_patterns.hy ..                          [ 66%]
tests/native_tests/native_macros.hy .......................      [ 70%]
tests/native_tests/operators.hy ................................ [ 76%]
........................                                         [ 80%] 
tests/native_tests/py35_only_tests.hy .........                  [ 81%]
tests/native_tests/py36_only_tests.hy ...                        [ 82%]
tests/native_tests/py3_only_tests.hy .......                     [ 83%]
tests/native_tests/quote.hy ..........                           [ 85%]
tests/native_tests/tag_macros.hy ..........                      [ 87%]
tests/native_tests/unless.hy .                                   [ 87%]
tests/native_tests/when.hy .                                     [ 87%]
tests/native_tests/with_decorator.hy ....                        [ 88%]
tests/native_tests/with_test.hy .....                            [ 88%]
tests/native_tests/contrib/hy_repr.hy ...........                [ 90%]
tests/native_tests/contrib/loop.hy ...                           [ 91%]
tests/native_tests/contrib/multi.hy .......                      [ 92%]
tests/native_tests/contrib/sequences.hy .....                    [ 93%]
tests/native_tests/contrib/walk.hy .........................     [ 97%]
tests/native_tests/extra/anaphoric.hy ............               [ 99%]
tests/native_tests/extra/reserved.hy .                           [100%]
```
</details>

```
=============================== FAILURES ===============================
_______________________________ test_doc _______________________________

capsys = <_pytest.capture.CaptureFixture object at 0x7f47d14353c8>

>   ;; Copyright 2019 the authors.
E   NameError: name '__macros__' is not defined

tests/native_tests/core.hy:1: NameError
=========== 1 failed, 577 passed, 1 xfailed in 8.18 seconds ============
```

--- a/tests/native_tests/core.hy
+++ b/tests/native_tests/core.hy
@@ -700,8 +700,12 @@ result['y in globals'] = 'y' in globals()")
                         <p> Move along. (Nothing to see here.)</p>)))
 
 (defn test-doc [capsys]
-  (doc doc)
-  (setv out_err (.readouterr capsys))
-  (assert (.startswith (.strip (first out_err))
-            "Help on function doc in module hy.core.macros:"))
-  (assert (empty? (second out_err))))
+  (defmacro testmacdoc []
+    "a fancy docstring"
+    '(+ 2 2))
+  (doc testmacdoc)
+  (setv [out err] (.readouterr capsys))
+  (assert (.startswith (.strip out)
+            "Help on function testmacdoc in module "))
+  (assert (in "a fancy docstring" out))
+  (assert (empty? err)))

